File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string temp = "";

        for(char ch: s) {
            if(isalnum(ch)) {
                temp += tolower(ch);
            }
        }

        int low = 0;
        int high = temp.length()-1;
        while(low<=high) {
            if(temp[low]!=temp[high]) {
                return false;
            }
            low++;
            high--;
        }

        return true;
    }
};

File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {

        String temp = "";

        for(char ch: s.toCharArray()) {
            if(Character.isLetterOrDigit(ch)) {
                temp += Character.toLowerCase(ch);
            }
        }
        int i = 0;
        int j = temp.length()-1;

        while(i <= j) {
            if(temp.charAt(i) == temp.charAt(j)) {
                i++;
                j--;
            }
            else {
                return false;
            }
        }
        
        return true;
    }
}